roman-numerals-solution/RomanNumeralsConvert.java

```diff
@@ -0,0 +1,167 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class RomanNumeralsConvert {
+	
+	public final static String onesArray[] = { "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX" };
+	public final static String tensArray[] = { "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC" };
+	public final static String hundredsArray[] = { "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM" };
+
+	public final static String TEST_MOD = "-test";
+	public final static String ONE = "I";
+	public final static String TWELVE = "XII";
+	public final static String ONE_HUNDRED_AND_THREE = "CIII";
+
+	private static String[] numArray;
+
+	public static void main(String[] args) {
+
+		if (args.length > 0) {
+			if (args.length > 1 || !TEST_MOD.equals(args[0])) {
+				System.out.println("Only one flag supports: - test");
+			} else {
+				runTest();
+			}
+		} else {
+			readInput();
+
+			validate();
+
+			convert();
+
+			printResult();
+		}
+
+	}
+
+	public static void readInput() {
+		try (BufferedReader in = new BufferedReader(new InputStreamReader(System.in));) {
+			String line;
+			while ((line = in.readLine()) != null) {
+				line = line.replace("\"", "").trim();
+				numArray = line.split("\\s+");
+			}
+		} catch (final IOException e) {
+			System.err.println("IOException reading System.in" + e);
+		}
+	}
+
+	public static void validate() {
+		for (int i = 0; i < numArray.length; i++) {
+			final String numStr = numArray[i];
+			boolean e = true;
+			for (int k = 0; k < numStr.length(); k++) {
+				if (Character.isDigit(numStr.charAt(k)) == false) {
+					numArray[i] = numArray[i] + ", illegal argument error - is not valid digital number.";
+					e = false;
+					break;
+				}
+			}
+
+			if (e) {
+				final int num = Integer.parseInt(numStr);
+				if (num > 3999 || num < 1) {
+					numArray[i] = numArray[i] + ", range bound error - number should be in the range [1 - 3999].";
+				}
+			}
+		}
+	}
+
+	public static void convert() {
+		for (int i = 0; i < numArray.length; i++) {
```

Comment:
In general, you should use collections, because there are way better functions to work with them than with arrays.